qemu/tests/rh_kernel_update.py

```diff
@@ -270,149 +147,150 @@ def upgrade_guest_pkgs(session, pkg, arch, debuginfo=False,
+                    % (pm_bin, kernel_pkg_dir))
+        status, output = session.cmd_status_output(inst_cmd,
+                                                   timeout=inst_timeout)
+        if status != 0:
+            test.fail("Failed to install kernel package(s): %s" % output)
+        session.cmd("rm -rf %s" % kernel_pkg_dir)
+
+    kernel_path = "/boot/vmlinuz-%s" % kernel_ver
+    if install_virtio:
+        error_context.context("Installing virtio driver")
         initrd_prob_cmd = "grubby --info=%s" % kernel_path
         s, o = session.cmd_status_output(initrd_prob_cmd)
         if s != 0:
-            msg = ("Could not get guest kernel information,"
-                   " guest output: '%s'" % o)
-            logging.error(msg)
-            test.error(msg)
-
+            test.error("Could not get guest kernel information: %s" % o)
         try:
             initrd_path = re.findall("initrd=(.*)", o)[0]
         except IndexError:
-            test.error("Could not get initrd path from guest,"
-                       " guest output: '%s'" % o)
-
-        driver_list = ["--with=%s " % drv for drv in virtio_drivers]
-        mkinitrd_cmd = "mkinitrd -f %s " % initrd_path
-        mkinitrd_cmd += "".join(driver_list)
-        mkinitrd_cmd += " %s" % kernel_version
+            test.error("Could not get initrd path from guest: %s" % o)
 
         error_context.context("Update initrd file", logging.info)
+        driver_list = ["--with=%s" % drv for drv in virtio_drivers]
+        mkinitrd_cmd = "mkinitrd -f %s " % initrd_path
+        mkinitrd_cmd += " ".join(driver_list)
+        mkinitrd_cmd += " %s" % kernel_ver
         s, o = session.cmd_status_output(mkinitrd_cmd, timeout=360)
         if s != 0:
-            msg = "Failed to install virtio driver, guest output '%s'" % o
-            logging.error(msg)
-            test.fail(msg)
+            test.fail("Failed to install virtio driver: %s" % o)
 
     # make sure the newly installed kernel as default
-    if ifupdatekernel:
+    if updated:
         error_context.context("Make the new installed kernel as default",
                               logging.info)
         make_def_cmd = "grubby --set-default=%s " % kernel_path
         s, o = session.cmd_status_output(make_def_cmd)
         if s != 0:
-            msg = ("Fail to set %s as default kernel,"
-                   " guest output: '%s'" % (kernel_path, o))
-            logging.error(msg)
-            test.error(msg)
+            test.error("Fail to set default kernel: %s" % o)
 
     # remove or add the required arguments
-
     error_context.context("Update the guest kernel cmdline", logging.info)
-    remove_args_list = ["--remove-args=%s " % arg for arg in args_removed]
+    remove_args_list = ["--remove-args=%s" % arg for arg in args_removed]
     update_kernel_cmd = "grubby --update-kernel=%s " % kernel_path
-    update_kernel_cmd += "".join(remove_args_list)
+    update_kernel_cmd += " ".join(remove_args_list)
```

Comment:
Prefer to keep the old format, add blank at the end of every  `--remove-args=%s `, and  no need to use blank when do  `" ".join`.  otherwise we need to add a blank before `--args` in L279.

current format is:
```
-remove-args=rhgb --remove-args=quiet--args="console=tty0 console=ttyS0,115200"
```
can not add "console=tty0 console=ttyS0,115200" correctly.

Reply:
Thanks for pointing out.